src/qgis_geonode/gui/search_result_widget.py

```
@@ -18,3 +25,62 @@ def __init__(self, geonode_resource: BriefGeonodeResource, parent=None):
         self.setupUi(self)
         self.name_la.setText(geonode_resource.title)
         self.description_la.setText(geonode_resource.abstract)
+        self.geonode_resource = geonode_resource
+
+        self.wms_btn.clicked.connect(self.wms_btn_clicked)
+        self.wfs_btn.clicked.connect(self.wfs_btn_clicked)
+
+    def wms_btn_clicked(self):
+        self.load_ogc_layer('wms')
+
+    def wfs_btn_clicked(self):
+        self.load_ogc_layer('wfs')
+
+    def load_ogc_layer(self, provider_key: str):
+        layer_uri = self.construct_ogc_uri(self.geonode_resource, provider_key)
+
+        if layer_uri == '':
+            return
```

This should show an error to the user - As a general rule, don't swallow errors silently